app/views/play/level/PlayLevelView.coffee

```diff
@@ -197,17 +197,42 @@ module.exports = class PlayLevelView extends RootView
       })
     @setupGod() if @waitingToSetUpGod
 
-    if features.china and e.level.get('slug') is 'magic-rush'
-      @checkForTournamentEnd()
+    if features.china
+      @levelSlug = e.level.get('slug')
+      @setupMandateCheck()
 
-  checkForTournamentEnd: =>
+  setupMandateCheck: =>
     return if @destroyed
     $.get '/db/mandate', (data) =>
       return if @destroyed
-      if data?[0]?.currentTournament isnt 'magic-rush'
-        window.location.href = '/play/ladder/magic-rush'
-      else
-        setTimeout @checkForTournamentEnd, 60 * 1000
+      courseInstanceID = @courseInstanceID or utils.getQueryVariable 'league'
+      if @isTournamentBlocked data?[0], courseInstanceID, @levelSlug
+        unless me.isAdmin()
+          window.location.href = '/play/ladder/'+@levelSlug+(if courseInstanceID then '/course/'+courseInstanceID else "")
+      setTimeout @setupMandateCheck, 60 * 1000
+
+  isTournamentBlocked: (mandate, courseInstanceID, levelSlug) =>
+    return unless mandate
+    tournament = _.find mandate?.currentTournament or [], (t) =>
+      t.courseInstanceID is courseInstanceID and t.level is levelSlug
+    if tournament
+      currentTime = @getCurrentDate() / 1000
+      return true unless tournament.startAt <= currentTime and tournament.endAt >= currentTime
+      delta = tournament.endAt - currentTime
+      console.log "Tournament end time: #{new Date(tournament.endAt * 1000)}, Time left: #{parseInt(delta / 60 / 60) }:#{parseInt(delta / 60) % 60}:#{parseInt(delta) % 60}"
+    else
+      return true if levelSlug in (mandate?.tournamentOnlyLevels or [])
+    return false
+
+
+  getServerDate: ->
+    res = $.ajax {async: false}
```

Comment:
The same concern as what I mentioned above. We do not want to use a deprecated feature or block the main thread to do an ajax call. See comment above: https://github.com/codecombat/codecombat/pull/5699/files#r363975467